kroxylicious-filters/kroxylicious-simple-transform/src/main/java/io/kroxylicious/proxy/filter/simpletransform/Replacing.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.simpletransform;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.IllegalCharsetNameException;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.plugin.Plugin;
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+
+@Plugin(configType = Replacing.Config.class)
+public class Replacing implements ByteBufferTransformationFactory<Replacing.Config> {
+    public record Config(
+                         @JsonProperty String charset,
+                         @JsonProperty(required = true) String targetPattern,
+                         @JsonProperty String replacementValue,
+                         @JsonProperty String replaceFrom) {
+        public Config(@JsonProperty String charset, @JsonProperty(required = true) String targetPattern, @JsonProperty String replacementValue,
```

Comment:
nit: i think you can use a compact ctor.

Reply:
I'm probably being dense but I don't think I can sensibly, due to the defaulting of the charset. I can add an overload which doesn't include `charset`, and default to UTF_8 there, in the signature but the canonical ctor can still be invoked with a `null` charset.